src/lib/import-export/tests/export/exporters/default-exporter.test.ts

```diff
@@ -0,0 +1,201 @@
+import fs from 'fs';
+import fsPromises from 'fs/promises';
+import os from 'os';
+import archiver from 'archiver';
+import { DefaultExporter } from '../../../export/exporters/default-exporter';
+import { ExportOptions, BackupContents } from '../../../export/types';
+
+jest.mock( 'fs' );
+jest.mock( 'fs/promises' );
+jest.mock( 'os' );
+
+// Create a partial mock of the Archiver interface
+type PartialArchiver = Pick<
+	archiver.Archiver,
+	'pipe' | 'file' | 'directory' | 'finalize' | 'on' | 'abort'
+>;
+
+const createMockArchiver = (): jest.Mocked< PartialArchiver > => {
+	return {
+		pipe: jest.fn().mockReturnThis(),
+		file: jest.fn().mockReturnThis(),
+		directory: jest.fn().mockReturnThis(),
+		finalize: jest.fn().mockResolvedValue( undefined ),
+		on: jest.fn().mockReturnThis(),
+		abort: jest.fn(),
+	};
+};
+
+// Mock archiver module
+jest.mock( 'archiver', () => {
+	return jest.fn().mockImplementation( () => createMockArchiver() );
+} );
+
+describe( 'DefaultExporter', () => {
+	let exporter: DefaultExporter;
+	let mockBackup: BackupContents;
+	let mockOptions: ExportOptions;
+	let mockArchiver: jest.Mocked< PartialArchiver >;
+	let mockWriteStream: { on: jest.Mock; path: string };
+
+	beforeEach( () => {
+		mockBackup = {
+			backupFile: '/path/to/backup.tar.gz',
+			wpConfigFile: '/path/to/wp-config.php',
+			sqlFiles: [ '/tmp/studio_export_123/file.sql' ],
+			wpContent: {
+				uploads: [ '/path/to/wp-content/uploads/file1.jpg' ],
+				plugins: [ '/path/to/wp-content/plugins/plugin1' ],
+				themes: [ '/path/to/wp-content/themes/theme1' ],
+			},
+		};
+
+		mockOptions = {
+			sitePath: '/path/to/site',
+			backupFile: '/path/to/backup.tar.gz',
+			includes: {
+				uploads: true,
+				plugins: true,
+				themes: true,
+				database: true,
+			},
+		};
+
+		// Reset all mock implementations
+		jest.clearAllMocks();
+
+		mockArchiver = createMockArchiver();
+		( archiver as jest.MockedFunction< typeof archiver > ).mockReturnValue(
+			mockArchiver as unknown as archiver.Archiver
+		);
+		mockWriteStream = {
+			on: jest.fn(),
+			path: '/path/to/backup.tar.gz',
+		};
+		( fs.createWriteStream as jest.Mock ).mockReturnValue( mockWriteStream );
+		( fsPromises.unlink as jest.Mock ).mockResolvedValue( undefined );
+		( fsPromises.mkdtemp as jest.Mock ).mockResolvedValue( '/tmp/studio_export_123' );
+		( fsPromises.writeFile as jest.Mock ).mockResolvedValue( undefined );
+		( os.tmpdir as jest.Mock ).mockReturnValue( '/tmp' );
+
+		mockArchiver.finalize.mockImplementation( () => {
+			return new Promise< void >( ( resolve ) => {
+				// Simulate async finalize
+				setTimeout( () => {
+					mockWriteStream.on.mock.calls.find( ( call ) => call[ 0 ] === 'close' )[ 1 ]();
+					resolve();
+				}, 0 );
+			} );
+		} );
+		exporter = new DefaultExporter( mockBackup );
```

Comment:
I noticed this test case doesn't have any expectations.

Reply:
What do you mean by that? This line is part of the `beforeEach`, and on tests, we are expecting certain parts to have been called.

Reply:
Oh, forget my comment. I've just realized that this logic is under `beforeEach` 😅 .